src/status_im/contexts/contact/blocking/events.cljs

```diff
@@ -0,0 +1,85 @@
+(ns status-im.contexts.contact.blocking.events
+  (:require
+    [legacy.status-im.data-store.chats :as chats-store]
+    [re-frame.core :as re-frame]
+    [status-im.contexts.chat.messenger.messages.list.events :as message-list]
+    [taoensso.timbre :as log]
+    [utils.re-frame :as rf]
+    [utils.transforms :as transforms]))
+
+(rf/defn clean-up-chat
+  [{:keys [db]} public-key
+   {:keys [chat-id unviewed-messages-count unviewed-mentions-count last-message]}]
+  (let [ids (keep (fn [[message-id {:keys [from]}]]
+                    (when (= from public-key) message-id))
+                  (get-in db [:messages chat-id]))
+        db  (-> db
+                ;; remove messages
+                (update-in [:messages chat-id] #(apply dissoc % ids))
+                (update-in [:chats chat-id]
+                           assoc
+                           :unviewed-messages-count unviewed-messages-count
+                           :unviewed-mentions-count unviewed-mentions-count
+                           :last-message            last-message))]
+    {:db (assoc-in db
+          [:message-lists chat-id]
+          (message-list/add-many nil (vals (get-in db [:messages chat-id]))))}))
```

Comment:
Thanks for refactoring this code from legacy 🙌 

Side note: we may want to quickly double-check that blocking/unblocking contacts from the contact profile still works correctly.

Reply:
Yeah absolutely! We can come back and refactor more later on 🙏

Reply:
I also pulled down the code and manually tested the block/unblock logic, and all is well 🙌